### ISSUE TITLE
function meancoordinates miscalculate?

### ISSUE THREAD

**Issue by IvoPartschefeld** [qgis/QGIS#51824]
### What is the bug or the crash?

Does the function meancoordinates miscalculate? See video: https://youtu.be/qyu2L6bjiyw

### Steps to reproduce the issue

See Video

### Versions

3..26

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by raymondnijssen** [comment]
Tried with 1x1 square '(polygon((0 0, 0 1, 1 1, 1 0, 0 0))' and got (0.4 0.4) as a result. So I'm pretty sure it's using the first and last point twice.

**Comment by nicogodet** [comment]
@raymondnijssen 😃 

https://github.com/qgis/QGIS/blob/32c2cea54cb92bbb2243b222816c8154c2b9adf9/src/analysis/processing/qgsalgorithmmeancoordinates.cpp#L173-L174

**Comment by roya0045** [comment]
> @raymondnijssen 😃
> 
> https://github.com/qgis/QGIS/blob/32c2cea54cb92bbb2243b222816c8154c2b9adf9/src/analysis/processing/qgsalgorithmmeancoordinates.cpp#L173-L174

Yeah I think that skiping the first or last vertex if it's the same as the first one, is needed. If fact duplicating the first vertex is not even needed to have a polygon.
